cli/2024-10-cli-udt.md

```diff
@@ -0,0 +1,285 @@
+                                description: 'Environment name'
+                            queueName:
+                                type: 'string'
+                                description: 'Name of the queue'
+                            host:
+                                type: 'string'
+                                description: 'Hostname of the messaging service'        
+                            port:
+                                type: 'string'
+                                description: 'Port'
+                            connectionString:
+                                type: 'string'
+                                description: 'Connection string to the messaging service' 
+                        required: ['queueName', 'host', 'port', 'connectionString']
+                    capabilities: ["Recipe"]           
+    ....
+    ```
+
+1. Deb creates the resource type for the messaging service Plaid in Radius
+
+    ```bash
+    rad resource-type create -f plaid.yaml
+    ```
+
+2. Deb creates the resource instance for the messaging service Plaid
+
+    ```json
+    {
+        "name": "myplaid",
+        "type": "Mycompany.Messaging/plaidService",
+        "properties": {
+            "application": "myapp"
+            "environment": "dev"
+            "queueName": "myqueue"
+            "host": "plaidxyz.mycompany.com"
+            "port": "5672"
+    ...
+    }
+    ```
+
+    ```bash
+    rad resource create -f myplaid.json
+    ```
+    The resource `myplaid` of type `Mycompany.Messaging/plaidService`is created and available for use.
+
+4. Deb lists the resource types available in Radius
+
+    ```bash
+    rad resource-type list
+    ```
+    The list of resource types under the resource provider `Mycompany.Messaging` are displayed.
+
+5. Deb makes changes to the resource type definition of `Mycompany.Messaging/plaidService`
+
+    ```bash
+    rad resource-type create -f plaid.yaml
+    ```
+    Note that the experience for handling breaking changes and non-breaking changes are detailed in the [authoring experience feature spec]((https://github.com/radius-project/design-notes/pull/74)
+
+6. Deb wants to know if the resource-type is created successfully
+
+    ```bash
+    rad resource-type show Mycompany.Messaging/plaidService
+    ```
+    The details of the resource type `Mycompany.Messaging` are displayed.
+
+7. Deb deletes a resource-type Plaid from the resource provider Mycompany.Messaging
+
+    ```bash
+    rad resource-type delete Mycompany.Messaging/plaidService
+    Warning: `myapp` application has resources of type `Mycompany.Messaging/plaidResource` provisioned and running. Are you sure you want to delete the resource type plaidResource from the resource provider Mycompany.Messaging? (yes/no)
+    ```
+    The resource-type deletion errors out as there are resources provisioned and running under the resource type. The resource myplaid should be deleted before deleting the resource type. 
```

Comment:
I disagree with this behavior. Here is my use case:
1. Platform engineer creates Mycompany.Messaging/plaidService@2024-10-01
2. Developer create resources using the 2024-10-01 version
3. Platform engineer creates Mycompany.Messaging/plaidService@2024-11-15
4. Platform engineer wants to restrict new resources being created with version 2024-10-01 so deletes resource type Mycompany.Messaging/plaidService@2024-10-01
5. Existing resources remain deployed